test/hotspot/jtreg/vmTestbase/nsk/share/jpda/SocketIOPipe.java

```diff
@@ -219,23 +270,17 @@ protected void connect() {
         if (shouldStop)
             return;
 
-        connection = new SocketConnection(this, getName());
-
         if (listening) {
-            connection.setConnectingProcess(connectingProcess);
-            if (serverSocket == null) {
-                connection.bind(port, timeout);
-            } else {
-                connection.setServerSocket(serverSocket);
+            // listenerThread == null means the test is not updated yet
+            // to start IOPipe listening before launching debuggee.
```

Comment:
I like this workaround for non-updated tests. This spot can be simplified after all tests got updated.

Reply:
Yes, the plan is to put assert here